src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -57,24 +48,36 @@ public abstract class Renderer : IDisposable
         /// <param name="serviceProvider">The <see cref="IServiceProvider"/> to be used when initializing components.</param>
         public Renderer(IServiceProvider serviceProvider)
         {
+            if (serviceProvider is null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
             _componentFactory = new ComponentFactory(serviceProvider);
+            Dispatcher = NullDispatcher.Instance;
```

Comment:
What's the use case for `Dispatcher` being optional, versus a required constructor parameter?

Reply:
We didn't want to pay the price in WebAssembly

Reply:
I'm all in favour of `NullDispatcher` existing for that case. To clarify, my question was about it being an optional parameter.